packages/nuxt/src/pages/runtime/plugins/router.ts

```diff
@@ -252,6 +244,19 @@ const plugin: Plugin<{ router: Router }> = defineNuxtPlugin({
       await nuxtApp.callHook('page:loading:end')
     })
 
+    router.afterEach(async (to, _from) => {
+      if (to.matched.length === 0) {
+        await nuxtApp.runWithContext(() => showError(createError({
```

Comment:
If this process is registered before `beforeEach`, it will be executed before the global middleware. Therefore, it needs to be registered after `beforeEach`.

Reply:
is it sufficient to move it?

Reply:
Yes, simply moving this was sufficient. If there is no middleware that performs a redirect, a 404 is displayed. 
If there is middleware that performs a redirect, I have confirmed that the issue is resolved with the redirect.

This issue wasn't detected in the tests because [...slug].vue exists, so I'm considering fixing the tests, but I plan to do it in a separate PR.